graphql/src/test/java/com/linecorp/armeria/server/graphql/GraphqlServiceMultipartTest.java

```diff
@@ -0,0 +1,149 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.google.common.io.Files;
+
+import com.linecorp.armeria.client.BlockingWebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.ContentDisposition;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.graphql.scalar.MoreScalars;
+import com.linecorp.armeria.common.multipart.BodyPart;
+import com.linecorp.armeria.common.multipart.Multipart;
+import com.linecorp.armeria.common.multipart.MultipartFile;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.idl.errors.SchemaProblem;
+
+class GraphqlServiceMultipartTest {
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final File graphqlSchemaFile =
+                    new File(getClass().getResource("/multipart.graphqls").toURI());
+            final GraphqlService service =
+                    GraphqlService.builder()
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      c.scalar(MoreScalars.multipartFile());
+                                      final DataFetcher<String> fileUpload = fileUploadFetcher();
+                                      c.type("Mutation",
+                                             typeWiring -> typeWiring.dataFetcher("fileUpload", fileUpload));
+                                      final DataFetcher<List<String>> fileUploads = fileUploadsFetcher();
+                                      c.type("Mutation",
+                                             typeWiring -> typeWiring.dataFetcher("fileUploads", fileUploads));
+                                  })
+                                  .build();
+            sb.service("/graphql", service);
+        }
+    };
+
+    private static DataFetcher<String> fileUploadFetcher() {
+        return environment -> CompletableFuture.supplyAsync(() -> {
+            try {
+                final MultipartFile multipartFile = environment.getArgument("file");
+                return Files.asCharSource(multipartFile.file(), StandardCharsets.UTF_8).read();
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }, GraphqlServiceContexts.get(environment).blockingTaskExecutor()).join();
```

Comment:
`join()` might block an event loop.
Should we just return a `CompletableFuture<String>`? It looks an acceptable type. https://www.graphql-java.com/documentation/execution/#asynchronous-execution
```java
private static DataFetcher<CompletableFuture<String>> fileUploadFetcher() { ... }
```

Reply:
Good idea!